### PR TITLE
Add pagerduty notifications, email sends traceback instead of exception

### PR DESCRIPTION
**Pull request by oldmantaiter**

### CODE REVIEW

**Comment by oldmantaiter** [comment]
@mkhaitman If this looks good to you, lets merge and deploy to funk (and enable on those jobs)

File: inferno/lib/job.py [hunk 1]
```diff
@@ -152,16 +152,35 @@ def wait(self):
...
+                        self.job.name, mail_ex, exc_info=sys.exc_info())
+
+                if self.rule.notify_pagerduty:
+                    if not self.rule.notify_pagerduty_key:
+                        api_key = self.settings.get('pagerduty_api_key')
+                    else:
+                        api_key = self.rule.notify_pagerduty_key
+                    try:
+                        from inferno.lib.notifications import send_pagerduty
+                        send_pagerduty(job_id=self.job.name, job_fail=exc,
```
Comment (mkhaitman) [inline inferno/lib/job.py]: If the rule will still be re-attempted in an hour for example, the page would still go out. Intended behaviour?

Comment (oldmantaiter) [inline inferno/lib/job.py]: My thinking is that we should notify anyways, if the job is critical enough for a page - we'd want to know if it fails the first time (and any subsequent times).

File: inferno/lib/notifications.py [hunk 2]
```diff
@@ -27,3 +30,29 @@ def send_mail(job_id=None, job_fail=None, mail_to=None, mail_from=None, mail_ser
...
+def send_pagerduty(job_id=None, job_fail=None,
+                   api_key=None, retry=None, retry_delay=None):
+    api_url = "https://events.pagerduty.com/generic/2010-04-15/create_event.json"
+    if not job_id or not job_fail:
+        raise Exception(
+            "Missing job_id or job_fail for Pagerduty notification")
+    if not api_key:
+        raise Exception(
+            "Missing pagerduty API key")
+    if retry and retry_delay:
```
Comment (mkhaitman) [inline inferno/lib/notifications.py]: This retry stuff is un-used as part of this send_pagerduty method.  Should be removed or appended to the pd_data['details'] perhaps!

Comment (oldmantaiter) [inline inferno/lib/notifications.py]: @mkhaitman Agreed, added it to the description (can see it easier that way)

**Pull request merged**